.gitignore

```diff
@@ -1,23 +1,12 @@
 # Xcode
 #
-build/
-*.pbxuser
-!default.pbxuser
-*.mode1v3
-!default.mode1v3
-*.mode2v3
-!default.mode2v3
-*.perspectivev3
-!default.perspectivev3
-project.xcworkspace
-!default.xcworkspace
-xcuserdata
-*.xccheckout
-*.moved-aside
-DerivedData
-*.hmap
-*.ipa
-*.xcuserstate
+xcuserdata/
+
+**/*.xcodeproj/project.xcworkspace/*
```

Comment:
I'm not sure that it's a good idea to exclude everything of a workspace except the shared data. The internal workspace structure is arbitrary and Apple may change it at any time. It wouldn't be the first time they add new files that are then added for everyone using the project - potentially causing a confirmation dialog in their projects.
We should simply ignore the `xcuserdata`, but include all the other things.
For SPM support, we should also explicitly exclude the project generated by SPM and move our Xcode project into a subfolder (e.g. Xcode).

Reply:
@ffried I copied this content from WWDC 2019 ( [`BuildingCustomViewsInSwiftUI`](https://developer.apple.com/documentation/swiftui/drawing_and_animation/building_custom_views_in_swiftui) ).

Reply:
I don't know if it's clever to use a UI talk for `.gitignore` files 😉 
If you find some documentation from Apple recommending this, I'm fine with it. I'm just talking out of experience, that it's extremely cumbersome if e.g. the file with the 32-bit warning is ignored and devs get a confirmation dialog every time they open the project.

Reply:
@ffried 
I know one correct way to get ".gitignore" that Apple recommends - create repository for a new project. But.. I don't use Xcode for Git tasks :)

Reply:
Yea, but the `.gitignore` SPM generates is a lot simpler.